server/controllers/tripController.js

```diff
@@ -46,20 +45,65 @@ class TripController {
         DateOfTravel: trip.dataValues.dateOfTravel,
         DateOfReturn: req.body.returnDate,
       };
-      infoLogger(req, 200, message);
+
       util.setSuccess(200, message, data);
       return util.send(res);
+    
+    } catch(error){
+      const Error = `Internal Server Error + ${error}`;
+      return errorLogger(req, 500, Error); 
+    }
   }
 
   static async getAllCities(req, res){
+    try{
     const message = 'All cities supported by Barefoot Nomad';
 
     const data = await cityService.findAllCities();
-
-    infoLogger(req, 200, message);
+    
     util.setSuccess(200, message, data)
     return util.send(res);
+    
+     } catch(error){
+      const Error = `Internal Server Error + ${error}`;
+      return errorLogger(req, 500, Error);
+      }
+      
   }
+
+  static async updateOpenTripDetails (req, res){
+    try{
+      const newTripData = {
+        origin: req.body.origin || req.oldTripDetails.origin,
+        destination: req.body.destination || req.oldTripDetails.destination,
+        dateOfTravel: req.body.date || req.oldTripDetails.dateOfTravel,
+        dateOfReturn: req.body.returnDate || req.oldTripDetails.dateOfReturn,
+        travelReason: req.body.travelReason || req.oldTripDetails.travelReason,
+        accommodation_id: req.body.accommodationID || req.oldTripDetails.accommodation_id,
+        type: req.tripType,
+    }
+
+    const tripDetailsUpdater = await tripService.updateTrip(newTripData, { id: req.params.trip_id })
+    const message = `The ${newTripData.type} details were successfully updated`;
+    const data = {
+      Name: req.userData.fullName,
+      Email: req.userData.email,
+      From: tripDetailsUpdater[1].dataValues.origin,
+      Destination: tripDetailsUpdater[1].dataValues.destination,
+      DateOfTravel: tripDetailsUpdater[1].dataValues.dateOfTravel,
+      DateOfReturn: req.body.returnDate,
```

Comment:
You are returning returnDate from req.body, and what if for some reasons it is not one which is in database

Reply:
I return `returnDate` from the `req.body` in order to avoid this(in case of one way return):
![image](https://user-images.githubusercontent.com/52471122/79536047-5dbb9d80-807f-11ea-8b26-187eab6a4d25.png)

In addition, from my viewpoint, always the return date(and other update information) will be either the old trip data or the newly input data.

Reply:
I do not understand your first comment if you could make it clear for me it could be better.